### PR TITLE
Add logging to CyberSource signature generation

### PR DESCRIPTION
**Pull request by rlucioni**
This is meant to help us track where CyberSource data is being signed. The signature itself is not logged.

LEARNER-1671

This could help paint a more complete picture of the behavior we're seeing on the off-chance that Nginx access logs don't tell the whole story.

### CODE REVIEW

**Review by clintonb — APPROVED** [review]

File: ecommerce/extensions/payment/processors/cybersource.py [hunk 1]
```diff
@@ -273,6 +276,23 @@ def _generate_signature(self, parameters, use_sop_profile):
         Returns:
             unicode: the signature for the given parameters
         """
+        order_number = None
+        basket_id = None
+
+        if 'reference_number' in parameters:
```
Comment (rlucioni) [inline ecommerce/extensions/payment/processors/cybersource.py:282]: @clintonb this caught me off guard. The parameters we generate, sign, and send to the browser don't contain the `req_` prefix, but the parameters that come back from CyberSource do. I don't see us adding or removing that prefix anywhere in our code. How can the signature we send initially match the one we generate when handling CyberSource's response?

File: ecommerce/extensions/payment/processors/cybersource.py [hunk 2]
```diff
@@ -273,6 +276,23 @@ def _generate_signature(self, parameters, use_sop_profile):
...
+            order_number = parameters['reference_number']
+        elif 'req_reference_number' in parameters:
+            order_number = parameters['req_reference_number']
+
+        if order_number:
+            basket_id = str(OrderNumberGenerator().basket_id(order_number))
+
+        logger.info(
+            'Signing CyberSource payment data for basket [%s], to become order [%s].',
+            basket_id,
```
Comment (schenedx) [inline ecommerce/extensions/payment/processors/cybersource.py:292]: Does `logger.info` handle null basket_id and/or order_number?

**Review by schenedx — APPROVED** [review]
👍

File: ecommerce/extensions/payment/processors/cybersource.py (hunk 1 above)
Comment (clintonb) [reply ecommerce/extensions/payment/processors/cybersource.py:282]: See the signed_parameters field. The signature is generated based on that field's value. The keys used for the signature are not hard coded.

Comment (rlucioni) [reply ecommerce/extensions/payment/processors/cybersource.py:282]: @clintonb I see that, but we include the keys in the signed message. If the keys change after we sign the initial set of parameters, shouldn't that be a problem?

File: ecommerce/extensions/payment/processors/cybersource.py (hunk 2 above)
Comment (rlucioni) [inline ecommerce/extensions/payment/processors/cybersource.py:292]: @schenedx yes. The string representation is used: 'None'.

File: ecommerce/extensions/payment/processors/cybersource.py (hunk 1 above)
Comment (clintonb) [reply ecommerce/extensions/payment/processors/cybersource.py:282]: The keys may change, but the values should be the same. If they differ, the signatures should not match.

**Pull request merged**